clang/test/CodeGen/aie/aie2/aie2-scl2vec-intrinsic.cpp

```diff
@@ -435,7 +435,7 @@ v32uint16 test_insert(v32uint16 v, int idx, unsigned long long b) {
 
 // CHECK-LABEL: @_Z18test_broadcast_s16s(
 // CHECK-NEXT:  entry:
-// CHECK-NEXT:    [[CONV_I:%.*]] = sext i16 [[B:%.*]] to i32
+// CHECK-NEXT:    [[CONV_I:%.*]] = zext i16 [[B:%.*]] to i32
```

Comment:
Do you know why this changed to `zext`?
Since the parameter is a signed type, those are already sign extended (ABI requirement).
This change requires an explicit zero extension, although our intrinsic only reads the lower 16 bits.
Changing this to `anyext` would be ok, but zext seems wrong to me.

Reply:
In `SimplifyDemandedUseBits` a comment says that "if the NewBits are not demanded convert this into a zero extension." here "this" mean SExt. And also gaetan said "anyext does not exist in the middle end". So, maybe we need to let the SExt be, as this gets optimized later.

Reply:
zext seems to be preferred by demandedBits, probably because it yields a definite zero value for the upper bits.
Since the conversion is necessary to make the types sane, I think the conversion should be swallowed by instruction selection.

Reply:
Hi, with the change:
```
	nopb	;		nopa	;		nops	;		ret lr	;		nopm	;		nopv	
	nopx	                                //  Delay Slot 5
	nop	                                //  Delay Slot 4
	extend.u16	r0, r0                  //  Delay Slot 3
	vbcst.16	x0, r0                          //  Delay Slot 2
	nop	                                //  Delay Slot 1
```

Without:
```
	ret lr	
	nop	                                //  Delay Slot 5
	nop	                                //  Delay Slot 4
	nop	                                //  Delay Slot 3
	vbcst.16	x0, r0                          //  Delay Slot 2
	nop	                                //  Delay Slot 1
```

It seems that without we get rid of sext anyway.....

Reply:
Added a pattern for broadcast, now even zext got eliminated.

Reply:
Do you have some QoR results for this change?